backend/src/main/java/com/github/NicolaiKopka/web_page_services/user_pages/user_favorites/UserFavoritesService.java

```diff
@@ -30,23 +30,36 @@ public Collection<Movie> getAllFavoriteMoviesFromDbByUser(String username) {
 
         return userFavorites.getMovieIds().stream().map(movieDBApiConnect::getMovieById).toList();
     }
+
+    // TODO Returns immutable collections. Check why and fix
     public UserFavoritesSaveObject addMovieToFavorites(Integer movieId, String username) {
 
         MyUser user = myUserRepo.findByUsername(username).orElseThrow();
 
         try{
             UserFavoritesSaveObject userFavorites = userFavoritesRepo.findByUserId(user.getId()).orElseThrow();
-            // TODO any way to make return List mutable?
-            List<Integer> idList = new ArrayList<>(userFavorites.getMovieIds().stream().toList());
-            idList.add(movieId);
-            userFavorites.setMovieIds(idList);
+            List<Integer> movieIds = new ArrayList<>(List.copyOf(userFavorites.getMovieIds()));
+            if(movieIds.contains(movieId)) {
+                throw new IllegalArgumentException("Movie already in list");
+            }
+            movieIds.add(movieId);
+            userFavorites.setMovieIds(movieIds);
             return userFavoritesRepo.save(userFavorites);
         } catch (NoSuchElementException e) {
             UserFavoritesSaveObject userFavorites = new UserFavoritesSaveObject();
             userFavorites.setUserId(user.getId());
             userFavorites.setMovieIds(List.of(movieId));
             return userFavoritesRepo.save(userFavorites);
         }
+    }
 
+    public UserFavoritesSaveObject deleteMovieFromFavorites(int movieId, String username) {
+
+        MyUser user = myUserRepo.findByUsername(username).orElseThrow();
+        UserFavoritesSaveObject favoritesObject = userFavoritesRepo.findByUserId(user.getId()).orElseThrow();
+        List<Integer> idList = new ArrayList<>(List.copyOf(favoritesObject.getMovieIds()));
+        idList.remove(Integer.valueOf(movieId));
+        favoritesObject.setMovieIds(idList);
+        return userFavoritesRepo.save(favoritesObject);
```

Comment:
I would rewrite this method, so it returns an `Optional`. That way you can use the `ResponseEntity.of`-method.
```java
public Optional<UserFavoritesSaveObject> deleteMovieFromFavorites(int movieId, String username) {
    return myUserRepo.findByUsername(username)
            .flatMap(user -> userFavoritesRepo.findByUserId(user.getId()))
            .map(favoritesObject -> favoritesObject.deleteMovieId(movieId))
            .map(userFavoritesRepo::save)
}
```
And in `UserFavoritesSaveObject` you need a `deleteMovieId`-method:
```java
public void addMovieId(Integer movieId) {
    movieIds.removeIf(id -> Objects.equals(id, movieId));
    return this;
}
```

Reply:
But I think the way I implemented it now works. Was that what you meant?